pkgs/development/tools/literate-programming/noweb/default.nix

```diff
@@ -1,29 +1,74 @@
-{stdenv, fetchurl, gawk}:
-
-stdenv.mkDerivation {
-  name = "noweb-2.11b";
-  src = fetchurl {
-    urls = [ "http://ftp.de.debian.org/debian/pool/main/n/noweb/noweb_2.11b.orig.tar.gz"
-             "ftp://www.eecs.harvard.edu/pub/nr/noweb.tgz"
-          ];
-    sha256 = "10hdd6mrk26kyh4bnng4ah5h1pnanhsrhqa7qwqy6dyv3rng44y9";
+{ stdenv, fetchFromGitHub, gawk, gcc, groff, icon-lang ? null }:
+
+let noweb = stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  version = "2.12";
+  pname = "noweb";
+  tlType = "run";
+
+  src = fetchFromGitHub {
```

Comment:
Please provide proof this change is valid, preferably citing a source from the original url.

Reply:
What sort of proof do you want? Recent (as in the last couple years) development has been committed to the GitHub repo: https://github.com/nrnrnr/noweb

Reply:
As per the README:
> noweb is available from the Github repository https://github.com/nrnrnr/noweb

Reply:
I'm not asking for much as far as proof goes, maybe just a reference on the website to the github page. Just *something* so someone who has no domain specific knowledge of `noweb` can say "This is a legitimate change." as opposed to saying "I hope this PR isn't malicious."

Reply:
> As per the README:

That is more than enough. Thank you.

Reply:
Sorry, I spoke too soon. Can you link the README? I didn't see mention of github in it.

Reply:
It's on the second line: https://github.com/nrnrnr/noweb/blob/v2_12/README#L2-L3

Reply:
...but that's under the new url. It isn't in the README under the old url.

Is this a fork or continuation? If this is the same authors then there should be some mention on the site how to get the source, right?

Reply:
It's a continuation. I guess we could try to ping Norman. Here's a relevant commit, written by Norman, the author of noweb: https://github.com/nrnrnr/noweb/commit/a2c919eb74fffea9789bb3084e9312abcc4f65d4

Reply:
This seems plausible to me. The original URL identifies a user `nr` at Harvard. That user doesn't seem to exist anymore, but it seems clear this was one "Norman Ramsey". Today there is a Norman Ramsey at Tufts, whose Tufts website identifies himself as the creator of noweb. That actually links to https://www.cs.tufts.edu/~nr/noweb/index.html, and unfortunately I can't find any link to github.com/nrnrnr on his Tufts site, but it certainly seems like that's his GitHub account. If you want to confirm it you could email him at nr@cs.tufts.edu.

That said, AUR [cites github.com/nrnrnr](https://aur.archlinux.org/packages/noweb) as the upstream for its noweb package.

Reply:
Though maybe it's worth someone (ideally the OP here) dropping a line to Norman at his Tufts email suggesting that he should link to his GitHub on his page so as to prove that it's really his.

Reply:
Looking at it now I see it is the same author as the original so I think we've done our due diligence.

I'm not trying to be a bother, just encourage PR authors to include some sort of reference as a matter of precaution. Github has a number of planted Trojans so always good to be on the lookout!

Reply:
I understand and appreciate it! I'll email Norman when I can, and will report back.

Reply:
I’ve emailed Norman. One of us should have some update before too long.

Reply:
/cc @nrnrnr for maximum exposure/annoyance.

Reply:
I have updated the Noweb home page at Tufts.  

Thanks for distributing my work!

Reply:
Thank you!